describe/AbstractDataAttribute.ts

```diff
@@ -0,0 +1,112 @@
+import { DAType, DATypeDescription } from "./DAType.js";
+import { EnumType, EnumTypeDescription } from "./EnumType.js";
+import { NamingDescription, describeNaming } from "./Naming.js";
+import { ValDescription, describeVal } from "./Val.js";
+
+export function isAbstractDataAttributeDescription(
+  type: AbstractDataAttributeDescription
+): type is AbstractDataAttributeDescription {
+  return (
+    (type as AbstractDataAttributeDescription).bType !== undefined &&
+    (type as AbstractDataAttributeDescription).count !== undefined &&
+    (type as AbstractDataAttributeDescription).valImport !== undefined &&
+    (type as AbstractDataAttributeDescription).val !== undefined &&
+    (type as AbstractDataAttributeDescription).valKind !== undefined
```

Comment:
If these `!== undefined` things requiring type casts ever get too tedious for you to write, you could also use the retro-hip [in operator](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/in) and write these like so:

```suggestion
    'bType' in type && 'count' in type && 'valImport' in type && 'val' in type && 'valKind' in type
```

The semantic difference is that this would evaluate `true` for

```ts
const type = { bType: undefined, count: undefined, valImport: undefined, val: undefined, valKind: undefined }
```

since it only cares whether keys are present, not about their values.

Alternatively, if you truly care about the values as well, a stronger check would be possible by checking the types of the values instead of their merely not being `undefined`.

Reply:
Thank you for the hint! That does read better